File: Assets/ProjectGame/Lib/data/class_data_id/RoleType/RoleTypeTable.py
from ..BaseClassDataID import BaseClassDataID
from .RoleTypeRow import RoleTypeRow
from .RoleTypeTableID import RoleTypeTableID
from typing import Dict

class RoleTypeTable(BaseClassDataID):
    Table: Dict[RoleTypeTableID, RoleTypeRow] = {}

    @classmethod
    def _get_enum(cls, name: str):
        return RoleTypeTableID[name]
    @classmethod
    def _get_row_class(cls):
        return RoleTypeRow

    def read(self, reader):
        self.Table.clear()
        row_count = reader.read_int32()
        col_count = reader.read_int32()
        for _ in range(col_count):
            len_name = reader.read_int32()
            _ = reader.read_string()  # col name（ヘッダー読み飛ばし）
            len_type = reader.read_int32()
            _ = reader.read_string()  # col type
        for _ in range(row_count):
            enum_int = reader.read_int32()
            enum_val = RoleTypeTableID(enum_int)
            row = RoleTypeRow()
            row.read(reader)
            self.Table[enum_val] = row

File: Assets/ProjectGame/Lib/data/class_data_id/RoleType/RoleTypeTable.js
import { BaseClassDataID } from '../BaseClassDataID.js';
import { RoleTypeRow } from './RoleTypeRow.js';
import { RoleTypeTableID } from './RoleTypeTableID.js';

export class RoleTypeTable extends BaseClassDataID {
    static Table = new Map();

    static _getEnum(name) {
        return RoleTypeTableID[name];
    }

    static _getRowClass() {
        return RoleTypeRow;
    }

    read(reader) {
        this.constructor.Table.clear();
        const rowCount = reader.readInt32();
        const colCount = reader.readInt32();
        for (let i = 0; i < colCount; i++) {
            const lenName = reader.readInt32();
            reader.readString(); // col name（ヘッダー読み飛ばし）
            const lenType = reader.readInt32();
            reader.readString(); // col type
        }
        for (let r = 0; r < rowCount; r++) {
            const enumInt = reader.readInt32();
            const enumVal = RoleTypeTableID[Object.keys(RoleTypeTableID).find(key => RoleTypeTableID[key] === enumInt)] || enumInt;
            const row = new RoleTypeRow();
            row.read(reader);
            this.constructor.Table.set(enumVal, row);
        }
    }

    static getRow(id) {
        return this.Table.get(id) ?? null;
    }
}
